smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/ContentMD5Middleware.kt

```diff
@@ -1,26 +1,31 @@
 package software.amazon.smithy.swift.codegen.integration.middlewares
 
 import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.traits.HttpChecksumRequiredTrait
 import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
 import software.amazon.smithy.swift.codegen.SwiftWriter
 import software.amazon.smithy.swift.codegen.middleware.MiddlewarePosition
+import software.amazon.smithy.swift.codegen.middleware.MiddlewareRenderable
 import software.amazon.smithy.swift.codegen.middleware.MiddlewareStep
 import software.amazon.smithy.swift.codegen.model.capitalizedName
+import software.amazon.smithy.swift.codegen.model.hasTrait
 
-class ContentMD5Middleware {
-    val name = "ContentMD5Middleware"
+class ContentMD5Middleware : MiddlewareRenderable {
+    override val name = "ContentMD5Middleware"
 
-    val middlewareStep = MiddlewareStep.BUILDSTEP
+    override val middlewareStep = MiddlewareStep.BUILDSTEP
 
-    val position = MiddlewarePosition.BEFORE
+    override val position = MiddlewarePosition.BEFORE
 
     fun render(
         op: OperationShape,
         writer: SwiftWriter,
         outputShapeName: String,
         operationStackName: String
     ) {
-        val outputErrorName = "${op.capitalizedName()}OutputError"
-        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N<$outputShapeName, $outputErrorName>())", ClientRuntimeTypes.Middleware.ContentMD5Middleware)
+        if (op.hasTrait<HttpChecksumRequiredTrait>()) {
+            val outputErrorName = "${op.capitalizedName()}OutputError"
```

Comment:
can we use the static function `getOperationErrorShapeName` inside `ServiceGenerator` to get the name of an output error to keep consistency?